app/lib/link_details_extractor.rb

```diff
@@ -101,9 +101,17 @@ def root_array(root)
     end
 
     def json
-      @json ||= root_array(Oj.load(@data))
-                .map { |node| JSON::LD::API.compact(node, 'https://schema.org') }
-                .find { |node| SUPPORTED_TYPES.include?(node['type']) } || {}
+      @json ||= begin
+        nodes = root_array(Oj.load(@data)).map do |node|
+          begin
+            JSON::LD::API.compact(node, 'https://schema.org')
+          rescue JSON::LD::JsonLdError
+            node['type'] = node['@type']
```

Comment:
Why this change?

Reply:
Because the compacting converts `@type` to `type`. I don't understand how compacting works – I just assume this is intended. 

Prior to #31288 we were checking `@type`: https://github.com/mastodon/mastodon/commit/e9e63ed203996a4e06e31a7220e9d8a2df71f67c

Reply:
Note that we have decided to revert JSON-LD compacting for now. I'll explain in more details when I have more time, but basically it was creating way more issues than it was solving (and did not fix the original issue, which as not the presence of *a* named graph, but multiple ones)

Reply:
So, to expand a bit on this:
- the issue we tried to solve using JSON-LD compacting was a document having multiple named graphs, but compacting by itself does not help, we need to select the graph we are interested in; compacting would have worked if there was only one named graph but that was not the case
- compacting JSON-LD requires knowing the full context, which may require fetching remote documents; in particular, the `json-ld` gem will by default automatically fetch remote documents with its own networking code (that, in particular, does not have our timeouts or address range checks)
- the most correct approach would be to expand the JSON-LD document and check the exact properties we are interested in, but just as with compacting, this can require fetching remote documents, and is furthermore very brittle because it will not handle malformed documents well even though those can generally be assumed to have the expected shape when considering them as simple JSON